connectors/src/connectors/notion/lib/notion_api.ts

```diff
@@ -758,24 +758,44 @@ export async function retrieveDatabaseChildrenResultPage({
   }
 }
 
-export async function renderChildDatabaseFromPages({
+// This function is used to create a text representation of a notion database properties.
+// We use it to render databases inline (in the Notion Page document on Dust), and to create
+// structured Tables on Dust (we use the CSV format).
+// The function accepts a `dustIdColumn` array which must have the same length as the `pagesProperties`. This
+// array is used to add a column to the CSV that contains the Dust ID of the page (__dust_id). This is useful
+// to uniquely identify the notion page in the CSV.
+export async function renderDatabaseFromPages({
   databaseTitle,
   pagesProperties,
+  dustIdColumn,
   rowBoundary = "||",
   cellSeparator = " | ",
 }: {
   databaseTitle: string | null;
   pagesProperties: PageObjectProperties[];
+  dustIdColumn?: string[];
```

Comment:
We don't have access to the full "page cache entry" in this function (that's where the ID is coming from). We could pass the whole thing to it, do you think it would be better ? We'd still want a boolean because we don't want to add this column in all cases (we want it to be configurable)